configurations/rolling-upgrade-artifacts.yaml

```diff
@@ -0,0 +1,10 @@
+# workloads
+stress_before_upgrade: cassandra-stress write cl=QUORUM n=1000000 -schema 'keyspace=keyspace_entire_test replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=200 -pop seq=1..1000000 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5
+stress_during_entire_upgrade:
+  - cassandra-stress write cl=QUORUM n=1000000 -schema 'keyspace=keyspace_during_upgrade replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=200 -pop seq=1..1000000 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5
```

Comment:
```suggestion
  - cassandra-stress write cl=QUORUM n=50000000 -schema 'keyspace=keyspace_during_upgrade replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=200 -pop seq=1..50000000 -col 'n=FIXED(1) size=FIXED(1024)' -log interval=5
```

Reply:
We need to measure how long it takes, I aimed for something that will take at least few minutes to cover the upgrades and rollbacks.
if it's  too slow we can either increase number of threads or remove number of keys.

Reply:
Measured the suggested params - it takes ~46 minutes to execute writes and then it takes ~38 minutes to read that amount of rows at the end.
Repeated a couple times with lower values - 15M match well enough onto initial upgrade>rollback sequence. Also increased the number of threads in final read for this ks, so that we don't wait another 25 mins to finish the read.

Reply:
the tests duration raised a little. Test runs after adding the additional stress:
- ubuntu 2004, +16 mins, from 1h
- ubuntu 2204, +7 mins, from 58 mins
- centos9, +16mins, from 66 mins
- debian showed the same time, but its' only a single re-run, I suspect that in average its time will also raise a bit

Reply:
So, I don't understand where do we spend all this time.
My goal was to decrease the time, not increase it.
Can you please add a high level breakdown of the times?
I think we have the "steps" as INFO event, so maybe just include here the steps with their timings.


AFAIK the upgrade & rollbacks themselves should be very quick.
So, I want all the rest (write, reads, to be very short).

Reply:
We actually reduced execution time significantly by moving from `test_rolling_upgrade` test (4.5-5h long in avg) to `test_generic_cluster_upgrade` (1h in avg). Then we added this additional write stress in a new ks - it added time to upgrade itself + post upgrade read of the written data, in average it added 15 min.

These are timestamps of the scenario with this added write workload (for debian distro, but in average it is similar among other distros):
- initial prepare stress (1M rows) - 11:51:56->11:53:41 - `~2min`
>write stress during upgrade (15M rows) starts at 11:53:44
- initial upgrade of 2 nodes upgrade - 11:54:40->12:10:06 - `~16min`
- rollback of initially upgraded 2 nodes - 12:10:09->12:14:58 - `~15min`
>write stress during upgrade finishes at 12:15:32
- final upgrade of all 3 nodes - 12:15:08->12:28:42 - `~14min`
- post-upgrade read of data written pre-upgrade (1M rows) - 12:29:32->12:33:09 - `~4min`
- post-upgrade read of data written during upgrade (15M rows) - 12:29:31->12:41:47 - `~12min`

I also repeated couple runs to see how much time can be saved if:
- we don't do any stress during upgrade (as we have other configs, for ami, gce, etc. images, where we apply a few stress workloads during upgrade)
initial upgrade of 2 nodes - 15:12:03->15:26:32 - `~14 min`, no save
rollback nodes - 15:26:38->15:31:29 - `~5min`, 10min save
final upgrade - 15:31:36->15:45:17 - `~14 min`, no save
- we execute initial upgrade>rollback sequence for only one node (this one was repeated with no stress during upgrade, so should be compare with the numbers above)
initial upgrade - 16:21:21->16:28:46 - `~7min`, 7min save
rollback node - 16:28:51->16:31:14 - `~2.5min`, 2.5min save

These are single runs, so numbers can differ a bit next time, but gives some idea. 
@roydahan @fruch

Reply:
If we do need to have a stress during upgrade, then maybe one more option is to perform mixed stress, with e.g. `ratio(write=1,read=1)`.
This way we will move from 5 stress commands (1 write before, 2 parallel - write and read - during upgrade, and 2 parallel read after upgrade), to 3 commands (1 write before, 1 mixed write/read during upgrade, 1 read after upgrade).

Reply:
the slowest part in the upgrade of each node is
```
< t:2024-09-09 14:31:24,137 f:remote_base.py  l:560  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.19>: Running command "sudo DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confold" -o Dpkg::Options::="--force-confdef" upgrade -y "...
```

it takes ~4min for each node, so it's ~12m only on system upgrades (which varies between distros)
and then more than half of the upgrade, and it's has nothing todo with the stress.

I would say to go with no stress during, and with one rollback. it should get us below 1h, but I don't think we can make it much faster without starting to drop step from the documented sequence. (i.e. system upgrade)